FRONTEND/src/modules.js/navBar.js

```diff
@@ -0,0 +1,307 @@
+    change: function () {
+      updateNodes();
+    },
+    create: function () {
+      $("#yearAmount").val(
+        $("#year-slider-range").slider("values", 0) +
+        " - " +
+        $("#year-slider-range").slider("values", 1)
+      );
+    },
+  });
+  $("#occur-slider-range").slider({
+    range: true,
+    min: 0,
+    max: 100,
+    values: [20, 100],
+    slide: function (event, ui) {
+      $("#occurAmount").val(
+        logslider(ui.values[0]) + " - " + logslider(ui.values[1])
+      );
+    },
+    change: function () {
+      updateNodes();
+    },
+    create: function () {
+      $("#occurAmount").val(
+        logslider($("#occur-slider-range").slider("values", 0)) +
+        " - " +
+        logslider($("#occur-slider-range").slider("values", 1))
+      );
+    },
+  });
+};
+
+
+
+// ------------------------------ Function-7 ------------------------------
+function initAutocomplete(toolTopicData, addNodesFn, toolImage, databaseImage, topicImage) {
+  $("#tooltopic_autocomplete").autocomplete({
+    source: function (request, response) {
+      let term = $.ui.autocomplete.escapeRegex(request.term);
+      let matcher1 = new RegExp("^" + term, "i");
+      let matcher2 = new RegExp("^.+" + term, "i");
+      function subarray(matcher) {
+        return $.grep(toolTopicData, function (item) {
+          return matcher.test(item.value);
+        });
+      }
+      response($.merge(subarray(matcher1), subarray(matcher2)));
+    },
+    minLength: 1,
+    select: function (event, ui) {
+      let name = ui.item.value;
+      let idNode = ui.item.idNodes;
+      let labelNode = ui.item.labelnode;
+      if (Array.isArray(labelNode)) {
+        labelNode = labelNode[0];
+      }
+      addNodesFn(name, idNode, labelNode);
+      $(this).val("");
+      return false;
+    },
+    open: function () {
+      $(".ui-autocomplete").css("z-index", 1000);
+    },
+  }).autocomplete("instance")._renderItem = function (ul, item) {
+    if (item.labelnode[0] === "Tool") {
+      return $('<li><div class="boxAutocomplete"><img src="' +
+        toolImage +
+        '"><div><div class="TextAutocomplete">' +
+        item.value +
+        '</div><div class="typeSoft">' +
+        item.type.join("/") +
+        "</div></div></div></li>"
+      ).appendTo(ul);
+    } else if (item.labelnode[0] === "Database") {
+      return $('<li><div class="boxAutocomplete"><img src="' +
+        databaseImage +
+        '"><div><div class="TextAutocomplete">' +
+        item.value +
+        '</div><div class="typeSoft">' +
+        item.type.join("/") +
+        "</div></div></div></li>"
+      ).appendTo(ul);
+    } else {
+      return $('<li><div class="boxAutocomplete"><img src="' +
+        topicImage +
+        '"><div class="TextAutocomplete">' +
+        item.value +
+        "</div></div></li>"
+      ).appendTo(ul);
+    }
+  };
+}
+
+
+
+// ------------------------------ Function-8 ------------------------------
+function removeLegend() {
```

Comment:
do we need a const for this?
And need to be jQuery prob.

Reply:
Fixed!
const removed and use JQuery to select HTML element.